pandas/tseries/frequencies.py

```diff
@@ -464,7 +464,14 @@ def to_offset(freq):
         delta = None
         stride_sign = None
         try:
-            for stride, name, _ in opattern.findall(freq):
+            splitted = re.split(opattern, freq)
+            if splitted[-1] != '' and not splitted[-1].isspace():
+                # the last element must be blank
+                raise ValueError('includes garbage characters')
```

Comment:
this is an internal description which is catched immediately. Users wil see " invalid frequency: xxx" as an error.